projects/VRDR/DeathRecord_submissionProperties.cs

```diff
@@ -3906,15 +4235,115 @@ public string UsualIndustry
                 {
                     return;
                 }
+
+                Observation.ComponentComponent industryComponent= UsualWork.Component.FirstOrDefault(cmp => cmp.Code != null && cmp.Code.Coding != null && cmp.Code.Coding.Count() > 0 && cmp.Code.Coding.First().Code == "21844-6");
+                UsualWork.Component.RemoveAll(cmp => cmp.Code != null && cmp.Code.Coding != null && cmp.Code.Coding.Count() > 0 && cmp.Code.Coding.First().Code == "21844-6");
+                Dictionary<string, string>  usualIndustryDict = CodeableConceptToDict((CodeableConcept)industryComponent.Value);
+                usualIndustryDict["text"] = value;  // replace text, leave the rest alone
+                industryComponent.Value = DictToCodeableConcept(usualIndustryDict);
+                UsualWork.Component.Add(industryComponent);
+            }
+
+        }
+
+        /// <summary>Decedent's Usual Occupation (Coded).</summary>
+        /// <value>the decedent's usual occupation. A Dictionary representing a code, containing the following key/value pairs:
+        /// <para>"code" - the code</para>
+        /// <para>"system" - the code system this code belongs to</para>
+        /// <para>"display" - a human readable meaning of the code</para>
+        /// </value>
+        /// <example>
+        /// <para>// Setter:</para>
+        /// <para>Dictionary&lt;string, string&gt; mserv = new Dictionary&lt;string, string&gt;();</para>
+        /// <para>mserv.Add("code", "13-2011");</para>
+        /// <para>mserv.Add("system", VR.CodeSystems.IndustryDCCensus2018);</para>
+        /// <para>mserv.Add("display", "Accountants and Auditors");</para>
+        /// <para>ExampleDeathRecord.UsualOccupationCoded = uind;</para>
+        /// <para>// Getter:</para>
+        /// <para>Console.WriteLine($"UsualIndustryCoded: {ExampleDeathRecord.UsualIndustryCoded['display']}");</para>
+        /// </example>
+        [Property("Usual Occupation (Coded)", Property.Types.Dictionary, "Decedent Demographics", "Decedent's Usual Occupation.", true, ProfileURL.DecedentUsualWork, true, 40)]
+        [FHIRPath("Bundle.entry.resource.where($this is Observation).where(code.coding.code='21843-8')", "")]
+        public Dictionary<string, string> UsualOccupationCoded
+        {
+            get
+            {
+                if (UsualWork != null && UsualWork.Value != null && UsualWork.Value as CodeableConcept != null)
+                {
+                    return CodeableConceptToDict((CodeableConcept)UsualWork.Value);
+                }
+                return EmptyCodeableDict();
+            }
+            set
+            {
+                if (value == null || IsDictEmptyOrDefault(value))
+                {
+                    return;
+                }
+                if (UsualWork == null)
+                {
+                    CreateUsualWork();
+                }
+                UsualWork.Value = DictToCodeableConcept(value);
```

Comment:
This approach means that if someone 1) sets the UsualOccupation property with a string and then 2) sets just the code using the UsualOccupationCoded property setter then 3) the string that was previously set will be overwritten with nothing even though the user was not setting the string value. I don't know that this is an issue in practice or actually worth addressing in some way, but it is different than what I sketched out in email.

Reply:
Take a look now.  I think I've consistently addressed the issue, and beefed up the test cases to back up that claim.